std/haxe/math/bigint/BigInt.hx

```diff
@@ -0,0 +1,450 @@
+/*
+ * Copyright (C)2005-2022 Haxe Foundation
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package haxe.math.bigint;
+
+import haxe.ds.Vector;
+import haxe.io.Bytes;
+
+/* Original code courtesy Chuck Batson (github.com/cbatson) */
+@:allow(haxe.math.bigint)
+abstract BigInt(BigInt_) {
```

Comment:
Would it be possible in case of Java to delegate to the BigInteger class? Mabye other targets also have native BigInt implementations that could be used,

Reply:
Yes, that maybe will be a good solution as Java implementation have a very optimize methods for modPow and pow . The problems here is about some methods that not exist in Java, and need to be reimplemented here ( withouth using the current BigInt_.hx class). Most methods exists, but miss some for random and random prime numbers.

As I mention in above comments , there is also native BigInteger implementation in Python and Javascript which should be used here too. Native javascript implementation give better performance, but as the Java many current methods should be rewritten as Javascript implementation doesn't have any, 

C# also have BigInteger class , but I think the performance is not great. 

However, in some tests the current BigInteger have same or better performance against Java ( when use some of those benchmarks : https://github.com/maitag/littleBigInt/tree/master/benchmarks ) . Also Karatsuba multiplication in native Java is slower than the current implementation ( but the have a slighty faster one using implementation of Toom–Cook multiplication )

At the moment , I can't apply native BigInteger implementation for Java, C#, Javascript or Python , but hope so someone else will add them in the future. 

For me , the bigger drawback for Haxe implementation is the slower modPow ( and partly pow ) operations which is used for RSA .